targets/TARGET_NUVOTON/TARGET_M480/crypto/crypto-misc.cpp

```diff
@@ -170,42 +141,47 @@ void crypto_zeroize32(uint32_t *v, size_t n)
 
 bool crypto_aes_acquire(bool blocking)
 {
-    return crypto_submodule_acquire(&crypto_aes_sem, blocking);
+    return crypto_aes_mutex->trylock_for(blocking ? osWaitForever : 0);
 }
 
 void crypto_aes_release(void)
 {
-    crypto_submodule_release(&crypto_aes_sem);
+    crypto_aes_mutex->unlock();
 }
 
 bool crypto_des_acquire(bool blocking)
 {
-    return crypto_submodule_acquire(&crypto_des_sem, blocking);
+    return crypto_des_mutex->trylock_for(blocking ? osWaitForever : 0);
```

Comment:
Use of `osWaitForever` isn't documented with `trylock_for`, so I'd rather this was split into separate `lock` and `trylock_for` cases, if you really want the nonblocking version.

Do you really need the nonblocking version?

Reply:
@kjbracey-arm I remove the `blocking` parameter. As suggested, blocking or not depends on lock scope. For non-SHA AC, it is non-blocking to support `trylock` and fallback to S/W SHA on failure; for SHA AC, it is blocking to acquire AC for a short sequence of operations.